pyriemann/transfer/_rotate.py

```diff
@@ -0,0 +1,221 @@
+
+
+def _warm_start(X, Y, weights, metric='euclid'):
+    """Smart initialization of the minimization procedure.
+
+    The loss function being optimized is a weighted sum of loss functions with
+    the same structure and for which we know the exact analytic solution [1]_.
+
+    As such, a natural way to warm start the optimization procedure is to list
+    the minimizers of "local" loss function and set as Q0 the matrix that
+    yields the smallest value for the "global" loss function.
+
+    Note that the initialization is the same for both 'euclid' and 'riemann'.
+
+    References
+    ----------
+    .. [1] `Procrustes problems in Riemannian manifolds of positive definite
+        matrices
+        <https://hal.archives-ouvertes.fr/hal-02023293>`_
+        R. Bhatia and M. Congedo, Linear Algebra and its Applications,
+        Elsevier, 2019, 563, pp.440-445.
+    """
+
+    # obtain the solution of the local loss function
+    def _get_local_solution(Xi, Yi):
+        wX, qX = np.linalg.eig(Xi)
+        idx = wX.argsort()[::-1]
+        qX = qX[:, idx]
+        wY, qY = np.linalg.eig(Yi)
+        idx = wY.argsort()[::-1]
+        qY = qY[:, idx]
+        Qstar = qX @ qY.T
+        return Qstar
+
+    Q0_candidates, losses = [], []
+    for i in range(len(X)):
+        Q0_candidates.append(_get_local_solution(X[i], Y[i]))
+        losses.append(_loss(Q0_candidates[i], X, Y, weights, metric=metric))
+
+    i_min = np.argmin(losses)
+
+    return Q0_candidates[i_min]
+
+
+def _run_minimization(Q_ini, M_source, M_target, weights=None, metric='euclid',
+                      tol_step=1e-9, maxiter=10_000, maxiter_linesearch=32):
+
+    Q_1 = Q_ini
+
+    # loop over iterations
+    for _ in range(maxiter):
+
+        # get the current value for the loss function
+        F_1 = _loss(Q_1, M_target, M_source, weights, metric=metric)
+
+        # get the direction of steepest descent
+        direction = _project(
+            Q_1, _grad(Q_1, M_target, M_source, weights, metric=metric)
+        )
+
+        # backtracking line search
+        alpha = 1.0
+        tau = 0.50
+        r = 1e-4
+        Q = _retract(Q_1, -alpha * direction)
+        F = _loss(Q, M_target, M_source, weights, metric=metric)
+        for _ in range(maxiter_linesearch):
+            if F_1 - F > r * alpha * np.linalg.norm(direction)**2:
+                break
+            alpha = tau * alpha
+            Q = _retract(Q_1, -alpha * direction)
+            F = _loss(Q, M_target, M_source, weights, metric=metric)
+
+        # test if the step size is small
+        crit = np.linalg.norm(-alpha * direction)
+        if crit <= tol_step:
+            break
+
+        # update variables for next iteration
+        Q_1 = Q
+        F_1 = F
+
+    else:
+        warnings.warn('Convergence not reached.')
+
+    return Q, F
+
+
+def get_rotation_matrix(M_source, M_target, weights=None, metric='euclid',
```

Comment:
this function is not present in the API page. It means to me that it is meant to be private and therefore be called `_get_rotation_matrix`